espnet/nets/pytorch_backend/transformer/encoder_layer.py

```diff
@@ -6,6 +6,7 @@
 
```

Comment:
I feel that this part is very risky as we have a lot of "if", and might introduce some bugs. OK for now, but we could consider to refactor this and use the inheritance from transformer instead (as @hirofumi0810 and @ShigekiKarita suggested) in the later PR.

Reply:
Now we are exploring the improvement obtained by each part, such as `macaron style`, `cnn_module`. I think for a totally Conformer implement, we can remove the `if` of `macaron style`.

Reply:
Buggy here does not mean your conformer implementation.
This complicated `if` logic may break existing transformer implementations if someone else (not you) touch this part, which we should avoid.

Again, I think we should have `conformer` module inherited from the transformer module instead of modifying the transformer module. This can avoid complicated `if` logics and duplicated code, but we can do such refactoring in the later PR.

Reply:
Sorry, I found I misunderstand your and @hirofumi0810's suggestion. You mean to override the Transformer class, but I merged Conformer into Transformer.